hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/diskbalancer/DiskBalancerYaml.java

```diff
@@ -0,0 +1,152 @@
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package org.apache.hadoop.ozone.container.diskbalancer;
+
+import org.yaml.snakeyaml.DumperOptions;
+import org.yaml.snakeyaml.Yaml;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+import java.io.Writer;
+import java.nio.charset.StandardCharsets;
+
+/**
+ * Class for creating diskbalancer.info file in yaml format.
+ */
+
+public final class DiskBalancerYaml {
+
+  private DiskBalancerYaml() {
+    // static helper methods only, no state.
+  }
+
+  /**
+   * Creates a yaml file to store DiskBalancer info.
+   *
+   * @param diskBalancerInfo {@link DiskBalancerInfo}
+   * @param path            Path to diskBalancer.info file
+   */
+  public static void createDiskBalancerInfoFile(
+      DiskBalancerInfo diskBalancerInfo, File path)
+      throws IOException {
+    DumperOptions options = new DumperOptions();
+    options.setPrettyFlow(true);
+    options.setDefaultFlowStyle(DumperOptions.FlowStyle.FLOW);
+    Yaml yaml = new Yaml(options);
+
+    try (Writer writer = new OutputStreamWriter(
+        new FileOutputStream(path), StandardCharsets.UTF_8)) {
+      yaml.dump(getDiskBalancerInfoYaml(diskBalancerInfo), writer);
+    }
+  }
+
+  /**
+   * Read DiskBalancerConfiguration from file.
+   */
+  public static DiskBalancerInfo readDiskBalancerInfoFile(File path)
+      throws IOException {
+    DiskBalancerInfo diskBalancerInfo;
+
+    try (FileInputStream inputFileStream = new FileInputStream(path)) {
+      Yaml yaml = new Yaml();
+      DiskBalancerInfoYaml diskBalancerInfoYaml;
+      try {
+        diskBalancerInfoYaml =
+            yaml.loadAs(inputFileStream, DiskBalancerInfoYaml.class);
+      } catch (Exception e) {
+        throw new IOException("Unable to parse yaml file.", e);
+      }
+
+      diskBalancerInfo = new DiskBalancerInfo(
+          diskBalancerInfoYaml.isShouldRun(),
+          diskBalancerInfoYaml.getThreshold(),
+          diskBalancerInfoYaml.getBandwidthInMB(),
+          diskBalancerInfoYaml.getParallelThread());
+    }
+
+    return diskBalancerInfo;
+  }
+
+  /**
+   * Datanode DiskBalancer Info to be written to the yaml file.
+   */
+  public static class DiskBalancerInfoYaml {
+    private boolean shouldRun;
+    private double threshold;
+    private long bandwidthInMB;
+    private int parallelThread;
+
+    public DiskBalancerInfoYaml() {
+      // Needed for snake-yaml introspection.
+    }
+
+    @SuppressWarnings({"parameternumber", "java:S107"}) // required for yaml
```

Comment:
is suppressing parameternumber really needed? There are only 4 params but the threshold is 6 or 7 AFAIR

Reply:
Removed annotion.
Will mark as resolved.